### ISSUE TITLE
java nullpointerexception com.google.android.gms.ads.AdSize.equals(java.lang.Object)

### ISSUE THREAD

**Issue by kadirkoca** [googleads/googleads-mobile-unity#987]
Unity 2019.2.0f1
Admob Plugin V3.18.2
Ad Resolver 1.2.124

i tried every way but i cannot find solution.

always same error 
java.lang.NullPointerException: Attempt to invoke virtual method 'boolean com.google.android.gms.ads.AdSize.equals(java.lang.Object)

**Comment by Mokalipilou** [comment]
Hi,
Same problem, additional infos:
I have this since updating from Unity 2018.3.8f1 to Unity 2019.2.0.f1 , without change to AdMob related code and without plugins update  (GMA was then 1.17.0)
(i updated to solve [this Google Console warnings](https://forum.unity.com/threads/google-play-api-warnings-started-today.676654/), not Admob related)

I since updated GoogleMobileAds and PlayServiveResolver to 3.18.2 and 1.2.124, the game keep crashing.

If i comment out lines requesting a Banner, the game starts and other type of ads (interstitial and reward) works fine.

I suspend the game release waiting for a fix.

>>>
-06 19:57:29.906: E/AndroidRuntime(17450): java.lang.NullPointerException: Attempt to invoke virtual method 'boolean com.google.android.gms.ads.AdSize.equals(java.lang.Object)' on a null object reference
08-06 19:57:29.906: E/AndroidRuntime(17450): 	at com.google.android.gms.internal.ads.zzwu.zza(com.google.android.gms:play-services-ads-lite@@18.1.1:248)
08-06 19:57:29.906: E/AndroidRuntime(17450): 	at com.google.android.gms.internal.ads.zzwu.zza(com.google.android.gms:play-services-ads-lite@@18.1.1:87)
08-06 19:57:29.906: E/AndroidRuntime(17450): 	at com.google.android.gms.ads.BaseAdView.loadAd(com.google.android.gms:play-services-ads-lite@@18.1.1:15)
08-06 19:57:29.906: E/AndroidRuntime(17450): 	at com.google.android.gms.ads.AdView.loadAd(com.google.android.gms:play-services-ads-lite@@18.1.1:18)
08-06 19:57:29.906: E/AndroidRuntime(17450): 	at com.google.unity.ads.Banner$5.run(Banner.java:278)
08-06 19:57:29.906: E/AndroidRuntime(17450): 	at android.os.Handler.handleCallback(Handler.java:873)
08-06 19:57:29.906: E/AndroidRuntime(17450): 	at android.os.Handler.dispatchMessage(Handler.java:99)
08-06 19:57:29.906: E/AndroidRuntime(17450): 	at android.os.Looper.loop(Looper.java:214)
08-06 19:57:29.906: E/AndroidRuntime(17450): 	at android.app.ActivityThread.main(ActivityThread.java:7050)
08-06 19:57:29.906: E/AndroidRuntime(17450): 	at java.lang.reflect.Method.invoke(Native Method)
08-06 19:57:29.906: E/AndroidRuntime(17450): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
08-06 19:57:29.906: E/AndroidRuntime(17450): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)

**Comment by stowy** [comment]
I recommend using Unity version 2019.1.11f1 until we determine the cause of
this. We believe it could be a Unity issue, as the Smart banner reference
is null even when proguard is off.

Regards

Sam

**Comment by adibacco** [comment]
Same thing here. 
Unity 2019.2.0f1
Admob Plugin V3.18.2
Ad Resolver 1.2.124

> 08-07 22:56:12.318 23010 23010 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke virtual method 'boolean com.google.android.gms.ads.AdSize.equals(java.lang.Object)' on a null object reference
> 08-07 22:56:12.318 23010 23010 E AndroidRuntime:        at com.google.android.gms.internal.ads.zzwu.zza(com.google.android.gms:play-services-ads-lite@@18.1.1:248)
> 08-07 22:56:12.318 23010 23010 E AndroidRuntime:        at com.google.android.gms.internal.ads.zzwu.zza(com.google.android.gms:play-services-ads-lite@@18.1.1:87)
> 08-07 22:56:12.318 23010 23010 E AndroidRuntime:        at com.google.android.gms.ads.BaseAdView.loadAd(com.google.android.gms:play-services-ads-lite@@18.1.1:15)
> 08-07 22:56:12.318 23010 23010 E AndroidRuntime:        at com.google.android.gms.ads.AdView.loadAd(com.google.android.gms:play-services-ads-lite@@18.1.1:18)
> 08-07 22:56:12.318 23010 23010 E AndroidRuntime:        at com.google.unity.ads.Banner$5.run(Banner.java:278)
> 08-07 22:56:12.318 23010 23010 E AndroidRuntime:        at android.os.Handler.handleCallback(Handler.java:873)
> 08-07 22:56:12.318 23010 23010 E AndroidRuntime:        at android.os.Handler.dispatchMessage(Handler.java:99)
> 08-07 22:56:12.318 23010 23010 E AndroidRuntime:        at android.os.Looper.loop(Looper.java:214)
> 08-07 22:56:12.318 23010 23010 E AndroidRuntime:        at android.app.ActivityThread.main(ActivityThread.java:7076)
> 08-07 22:56:12.318 23010 23010 E AndroidRuntime:        at java.lang.reflect.Method.invoke(Native Method)
> 08-07 22:56:12.318 23010 23010 E AndroidRuntime:        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
> 08-07 22:56:12.318 23010 23010 E AndroidRuntime:        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:965)

**Comment by venvw** [comment]
Hi, I found **temporary** solution:

1. Go to Assets/GoogleMobileAds/Platforms/Android/Utils.cs
2. And **modify** method **_GetAdSizeJavaObject_** to

```
public static AndroidJavaObject GetAdSizeJavaObject(AdSize adSize)
{
    if (adSize.IsSmartBanner)
    {
         //return new AndroidJavaClass(AdSizeClassName).GetStatic<AndroidJavaObject>("SMART_BANNER");
         return new AndroidJavaObject(AdSizeClassName, adSize.Width, adSize.Height).GetStatic<AndroidJavaObject>("SMART_BANNER");
    }
    else
    {
        return new AndroidJavaObject(AdSizeClassName, adSize.Width, adSize.Height);
    }
}
```

**Comment by luvjungle** [comment]
is it fixed in 3.18.3?

**Comment by DownAP** [comment]
having the same fucking problem with 3 different versions i've tried

**Comment by gimoneto** [comment]
It  seems that some kind of bug calling GetStatic in Unity 2019.2+ is confirmed and registered in the Issue Tracker: https://issuetracker.unity3d.com/issues/android-androidjavaclass-dot-getstatic-returns-null-when-being-built-to-android-device

**Comment by stowy** [comment]
Fixed in v4.0.0 just released.
